### PR TITLE
handle syncing transcriptions for documents with more than one edition

### PR DESCRIPTION
**Pull request by rlskoeser**
in this PR:
- revised the sync transcription code to pull author names from the TEI files in structured form (which I added) so that we can identify the correct footnote for documents with multiple transcriptions
- code to create goitein typed text footnotes if needed for transcriptions to be synced (I thought this had already been merged in and run! but it's showing as a change)
- adjusted plain text document download to only show for logged in users, since the bidi support in plain text is not good enough for public use
- first pass implementation for displaying multiple transcriptions with image on the document details page; I also adjusted the styles for the transcription div so it should match the height of the image div and be scrollable (although occurs to me now that probably isn't what we want on mobile!)

 notes:
 - my implementation for showing multiple transcriptions isn't great; it's hard to see where the second one starts, and I know it's there! wondering if we need a table of contents and anchor links (can you link to an anchor within a scrollable div?)

### CODE REVIEW

**Comment by blms** [comment]
> my implementation for showing multiple transcriptions isn't great; it's hard to see where the second one starts, and I know it's there! wondering if we need a table of contents and anchor links (can you link to an anchor within a scrollable div?)

I wonder if this might need Gissoo's input. My original thinking was using a slideshow functionality as in the original designs, but that makes more sense when each of the transcriptions neatly corresponds to one image. In the case of multiple versions of transcription that correspond to the same image, I can't think of a simple solution.

**Review by blms — COMMENTED** [review]
Code all looks good to me!

There are a couple of things I noticed and I'm not sure they're within the scope of this PR:

1. I think the way the multiple editions are being displayed here is fine for now, but I did notice that a lot of them are still not getting transcription content in HTML. See PGPIDs 848, 591, and 4495 for examples. I remember we discussed before but wasn't sure if that was supposed to be addressed here or not.
2. Looks like some of the TEI isn't getting line numbers parsed correctly. Check out PGPIDs 5555, 4740, and 5499 for examples. In the HTML it looks like some lines' `value` attribute is `None`.

**Comment by rlskoeser** [comment]
@blms thanks for flagging those things — I knew about the `None` problem and had a solution for it at one point, but may have stashed and forgotten it (my work on this feature has unfortunately been very piecemeal).

There is still some data work ongoing that's needed to resolve the remaining footnotes and transcriptions, which I think should be independent of the code (this script should handle the cases once the necessary db and TEI updates are made).

However, it would be good to generate a list of footnotes with old-style transcription to check — because there shouldn't be any left once the script updates and data changes are done.

**Comment by blms** [comment]
> However, it would be good to generate a list of footnotes with old-style transcription to check — because there shouldn't be any left once the script updates and data changes are done.

<details>
<summary>old style transcriptions to check</summary>

```
Edition of T-S K15.85 (PGPID 4585)
pk: 5546

Edition of T-S K15.85 (PGPID 4585)
pk: 5545

Edition of NLI 577.3/2 (PGPID 848)
pk: 2026

Edition of DK 333 (alt: 18) (PGPID 4495)
pk: 868

Edition and Translation of CUL Add.3414.1 (PGPID 5384)
pk: 605

Edition and Translation of ENA 2727.7b (PGPID 4740)
pk: 1026

Edition and Translation of T-S 18J4.18 + T-S NS J21 (PGPID 4732)
pk: 3982

Edition and Translation of NLI 577.3/6 (PGPID 5499)
pk: 2030

Edition and Translation of CUL Or.1080 J95 (PGPID 5555)
pk: 789

Edition and Translation of T-S Ar.41.81 (PGPID 5540)
pk: 5140

Edition and Translation of ENA NS 21.14 (PGPID 2691)
pk: 1470

Edition of Bodl. MS heb. d 74/41 (PGPID 591)
pk: 446

Edition of CUL Or.1080 J24 (PGPID 9121)
pk: 722

Edition of Bodl. MS heb. f 39/29 (PGPID 606)
pk: 553

Edition of ENA NS 13.1 + BL OR 5546.1 (PGPID 9804)
pk: 87

Edition of DK 333 (alt: 18) (PGPID 4495)
pk: 869

Edition of NLI 577.3/2 (PGPID 848)
pk: 2027

Edition of Bodl. MS heb. d 74/41 (PGPID 591)
pk: 445

Edition of CUL Or.1080 J24 (PGPID 9121)
pk: 720

Edition of Bodl. MS heb. f 39/29 (PGPID 606)
pk: 554

Edition of ENA NS 13.1 + BL OR 5546.1 (PGPID 9804)
pk: 89

Edition of Bodl. MS heb. f 29/69 + Bodl. MS heb. f 29/68 (PGPID 2493)
pk: 540

Edition of T-S 8J5.1 (PGPID 2142)
pk: 4909

Edition of Bodl. MS heb. f 29/69 + Bodl. MS heb. f 29/68 (PGPID 2493)
pk: 541

Edition of T-S 8J5.1 (PGPID 2142)
pk: 4907

Edition of T-S 8J10.16 (PGPID 1849)
pk: 4306

Edition of T-S 8J5.1 (PGPID 2142)
pk: 4910

Edition of T-S 8J5.1 (PGPID 2142)
pk: 4908

Edition of T-S 8J10.16 (PGPID 1849)
pk: 4305

Edition of ENA NS 21.14 (PGPID 2691)
pk: 1469
```
</details>

Generated with `Footnote.objects.filter(doc_relation__contains=Footnote.EDITION).filter(content__isnull=False).exclude(content__lines=None)`

**Review by kmcelwee — APPROVED** [review]

**Comment by rlskoeser** [comment]
@blms updated script to better handle the missing line numbers; also improved footnote identification logic and made the script idempotent so same footnote won't get updated by different files (which was happening in qa). Also tweaked the document transcription display — maybe good enough for Gissoo to look at and advise?  (BUT: I think the transcription div should not have a max-height when there is no image, and maybe not on mobile. Easy to fix? In scope for this PR?)

My local check has 5 documents left with the old-style transcription, which I think should be resolved once we handle the remaining data work (a few more transcriptions to identify and edition footnotes to update or create).

File: sitemedia/scss/components/_iiif.scss [hunk 1]
```diff
@@ -68,6 +68,13 @@
     // Transcription text
     .transcription {
         max-width: 60ch;
+        /* make a scrollable div; match height with image container */
+        max-height: 320px;
+        overflow: scroll;
+        @include breakpoints.for-tablet-landscape-up {
+            max-height: 680px;
+        }
+
```
Comment (blms) [inline sitemedia/scss/components/_iiif.scss:77]: CSS for max-height and overflow (disabled on mobile / with no image, based on class `.no-image`):

```suggestion
        @include breakpoints.for-tablet-landscape-up {
            /* make a scrollable div; match height with image container */
            max-height: 680px;
            overflow: scroll;
            /* no scrolling/max height when no image present */
            &.no-image {
                max-height: none;
                overflow: auto;
            }
        }
```

File: geniza/corpus/templates/corpus/snippets/document_transcription.html [hunk 2]
```diff
@@ -22,7 +22,17 @@ <h1 class="sr-only">
     {% if document.has_transcription %}
         <div class="transcription">
```
Comment (blms) [inline geniza/corpus/templates/corpus/snippets/document_transcription.html:23]: Could add a class to the div when there's no image to handle in CSS:

```suggestion
        <div class="transcription{% if not document.has_image %} no-image{% endif %}">
```

**Review by blms — APPROVED** [review]
All your changes look good to me! I made some suggestions if you wanted to implement the max-height adjustments. Not sure if it's totally in scope but it's very easy to do.

**Pull request merged**